core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -693,9 +693,9 @@ private[spark] class BlockManager(
    */
   private def getRemoteValues[T: ClassTag](blockId: BlockId): Option[BlockResult] = {
     val ct = implicitly[ClassTag[T]]
-    getRemoteBytes(blockId).map { data =>
+    getRemoteManagedBuffer(blockId).map { data =>
       val values =
-        serializerManager.dataDeserializeStream(blockId, data.toInputStream(dispose = true))(ct)
+        serializerManager.dataDeserializeStream(blockId, data.createInputStream())(ct)
```

Comment:
I'm pretty sure this is OK.  If you're getting remote bytes, you're never going to get a `BlockManagerManagedBuffer`.  We should probably add that as an assert, though, to make sure.

What still puzzles me, though, is why we there was ever a `dispose=true` here.  That is something I'd like to look at more still.  I have a hard time believing it could do anything useful, as its not guaranteed to get called in a `finally` or a taskCompletionListener etc.

Reply:
I reran my system tests with a build with the assert. No issues.